Allow skipping test cases that need Internet access

bdrung opened the pull request:
Support skipping all test cases that access the Internet by setting the environment variable `NO_INTERNET`. This is useful to make the test run reproducible and robust for future runs (to avoid breaking in case some random service on the Internet changes).

s0undt3ch commented on tests/unit/utils/markers/test_skip_if_no_remote_network.py:
```diff
@@ -22,3 +23,12 @@ def test_no_remote_network():
         assert skip_reason == "No internet network connection was detected"
+
+
+def test_remote_network_with_no_internet():
+    environ = os.environ.copy()
+    environ["NO_INTERNET"] = "1"
+    with mock.patch("os.environ", return_value=environ):
+        skip_reason = markers.skip_if_no_remote_network()
+        assert skip_reason is not None
+        assert skip_reason == "Environment variable NO_INTERNET is set"
```
```suggestion
def test_remote_network_with_no_internet_env_variable():
    with mock.patch.dict(os.environ, {"NO_INTERNET": "1"}):
        skip_reason = markers.skip_if_no_remote_network()
        assert skip_reason is not None
        assert skip_reason == "Environment variable NO_INTERNET is set"
```

s0undt3ch reviewed (changes requested):
Please add a `changelog/16.feature.rst` file explaining the new feature added.

Also, could you elaborate on the use case for this?

bdrung commented:
Added `changelog/16.feature.rst` and updated commit message plus MR description. We want to set this environment variable during test run in Debian packages. Test in Debian packages / autopkgtest should not access the Internet (for the reason mentioned).

s0undt3ch commented:
Thank You!
I need to sort why the windows CI pipeline is failing and once I do, I'll push out a release.

s0undt3ch commented:
Released on 1.3.0
